gcc/rust/checks/lints/rust-lint-scan-deadcode.h

```diff
@@ -60,15 +60,16 @@ class ScanDeadcode : public MarkLiveBase
 	    if (!implBlock->has_trait_ref ())
 	      {
 		rust_warning_at (
-		  function.get_locus (), 0,
+		  function.get_function_name ().get_locus (), 0,
 		  "associated function is never used: %<%s%>",
 		  function.get_function_name ().as_string ().c_str ());
 	      }
 	  }
 	else
 	  {
 	    rust_warning_at (
-	      function.get_locus (), 0, "function is never used: %<%s%>",
+	      function.get_function_name ().get_locus (), 0,
+	      "function is never used: %<%s%>",
 	      function.get_function_name ().as_string ().c_str ());
```

Comment:
I fixed error location only for check for unused functions.
Of course, other error location should be fixed in another PR.

Reply:
Thanks for doing that but you are correct the location info like this is a much bigger issue and outside of scope for your google summer of code project so this PR is more than enough.